File: Problem 2/Problem2.c
#include <stdio.h>
#define LIMIT 1000000

int main(void) {
  if (LIMIT <= 2) {
    printf("2");
    return 0;
  }

  int temp, fib1 = 2, fib2 = 8, fib_sum = 10;

  while (fib2 < LIMIT) {
    temp = fib2;
    fib2 = fib1 + 4 * fib2;
    fib1 = temp;

    fib_sum += fib2;
  }
  
  printf("%d\n", fib_sum);
  return 0;
}

File: Problem 2/Problem2.hs
fib_sum :: Int -> Int -> Int -> Int -> Int
fib_sum f1 f2 f_sum limit
  | f2 >= limit  = f_sum
  | otherwise    = fib_sum f2 (f1 + 4 * f2) (f_sum + f1 + 4 * f2)

main :: IO()
main
  | limit <= 2   = print("2")
  | limit <= 8   = print("10")
  | otherwise    = print(fib_sum 2 8 0 1000)
